pkgs/http/lib/src/browser_client.dart

```diff
@@ -64,6 +64,14 @@ class BrowserClient extends BaseClient {
     var completer = Completer<StreamedResponse>();
 
     unawaited(xhr.onLoad.first.then((_) {
+      final contentLengthHeader = xhr.responseHeaders['content-length'];
+      if (!RegExp(r'^\d+$').hasMatch(contentLengthHeader ?? '0')) {
+        completer.completeError(ClientException(
+          'Invalid content-length header [$contentLengthHeader].',
+          request.url,
+        ));
+        return;
+      }
```

Comment:
I would avoid the RegExp check against `'0'` in case of `null`. I'd also consider pulling out the `RegExp` construction so it only happens once.

[optional] Also, you could optionally move the variable to be defined with `case` and make it only in scope within the conditional.

```suggestion
      if (xhr.responseHeaders['content-length'] case final contentLengthHeader?
          when !digitRegex.hasMatch(contentLengthHeader)) {
        completer.completeError(ClientException(
          'Invalid content-length header [$contentLengthHeader].',
          request.url,
        ));
        return;
      }
```

Reply:
Done